modules/free/src/main/scala/doobie/free/kleisliinterpreter.scala

```diff
@@ -73,12 +74,15 @@ import doobie.free.callablestatement.{ CallableStatementIO, CallableStatementOp
 import doobie.free.resultset.{ ResultSetIO, ResultSetOp }
 
 object KleisliInterpreter {
-  def apply[M[_]: WeakAsync](logHandler: LogHandler[M]): KleisliInterpreter[M] =
-    new KleisliInterpreter[M](logHandler)
+  def apply[M[_]: WeakAsync](logHandler: LogHandler[M], customPrimitive: Option[Primitive[M]] = None): KleisliInterpreter[M] =
+    new KleisliInterpreter[M](logHandler, customPrimitive)
 }
 
 // Family of interpreters into Kleisli arrows for some monad M.
-class KleisliInterpreter[M[_]](logHandler: LogHandler[M])(implicit val asyncM: WeakAsync[M]) { outer =>
+class KleisliInterpreter[M[_]](logHandler: LogHandler[M], customPrimitive: Option[Primitive[M]] = None)(implicit val asyncM: WeakAsync[M]) { outer =>
+
+  private val _primitive = customPrimitive.getOrElse(Primitive.Default[M]())
+
```

Comment:
Just a suggestion here to consider:
```suggestion
class KleisliInterpreter[M[_]](logHandler: LogHandler[M], customPrimitive: Primitive[M])(implicit val asyncM: WeakAsync[M]) { outer =>

  def this(logHandler: LogHandler[M])(implicit val asyncM: WeakAsync[M]) =
    this(logHandler, Primitive.Default[M])
```
Now you can use `customPrimitive` directly.
I.e. in this case the overloaded constructor helps to avoid dealing with `Option`.